modules/gateway/main.tf

```diff
@@ -0,0 +1,166 @@
+data "azurerm_resource_group" "graphdb" {
+  name = var.resource_group_name
+}
+
+data "azurerm_subnet" "graphdb-gateway" {
+  name                 = var.gateway_subnet_name
+  resource_group_name  = var.resource_group_name
+  virtual_network_name = var.network_interface_name
+}
+
+data "azurerm_public_ip" "graphdb-gateway" {
+  name                = var.gateway_public_ip_name
+  resource_group_name = var.resource_group_name
+}
+
+data "azurerm_user_assigned_identity" "graphdb-gateway-tls" {
```

Comment:
we should revisit the use of data sources like this at some point, see [this](https://support.hashicorp.com/hc/en-us/articles/4414363057299-Best-Practice-depends-on-Meta-Argument).

Reply:
https://github.com/Ontotext-AD/terraform-azure-graphdb/pull/13